js/app.js

```diff
@@ -1,5 +1,21 @@
-angular.module('GiphyApp', [])
+var app = angular.module('GiphyApp', []);
 
-.run(function() {
-  console.log('App has loaded!');
-});
+app.controller('SearchCtrl', ['$scope', '$http', function($scope, $http) {
+
+  $scope.userSearch = '';
+
+  $scope.searchForGifs = function() {
+    $http.get('http://api.giphy.com/v1/gifs/search?', {
+      params: {
+        q: $scope.userSearch,
+        api_key: 'dc6zaTOxFJmzC'
+      }
+    })
+    .then(function success(res) {
+      $scope.giphyResults = res.data.data;
+      $scope.userSearch = '';
+    }, function error(res) {
+      console.log(res);
+    });
+  };
```

Comment:
Here is some code from my deliverable to handle the infinite scrolling:

First I added the angular module 'infinite-scroll' (https://sroze.github.io/ngInfiniteScroll/):

    var giphyApp = angular.module('GiphyApp', ['infinite-scroll']);

And then the JS:

    $scope.search = function () {
        if ($scope.searchTerm != $scope.oldSearchTerm) {$scope.giphys = [];};
        var req = {
            url: "http://api.giphy.com/v1/gifs/search",
            method: 'GET',
            params: {
                api_key: $scope.api,
                offset: $scope.offset,
                q: $scope.searchTerm,
            }
        }
        $http(req).then(function success(res) {
            $scope.giphys = $scope.giphys.concat(res.data.data);
        }, function error (res) {
            console.log(res);
        });
    };

    $scope.myPagingFunction = function (){
        $scope.oldSearchTerm = $scope.searchTerm;
        $scope.offset = 25 * $scope.count;
        $scope.search();
        $scope.count++;
    };